packages/preview-middleware/templates/test/qunit.js

```diff
@@ -1,8 +1,28 @@
-sap.ui.define([
-    'sap/ui/thirdparty/qunit-2'
-], function(QUnit) {
-        QUnit.config.autostart = false;
-        sap.ui.require(<%- JSON.stringify(tests) %>, function() {
-            QUnit.start();
-    });
+sap.ui.loader.config({
+    shim: {
+        "sap/ui/qunit/qunit-junit": {
+            deps: ["sap/ui/thirdparty/qunit-2"]
+        },
+        "sap/ui/qunit/qunit-coverage": {
+            deps: ["sap/ui/thirdparty/qunit-2"]
+        },
+        "sap/ui/thirdparty/sinon-qunit": {
+            deps: ["sap/ui/thirdparty/qunit-2", "sap/ui/thirdparty/sinon"]
+        },
+        "sap/ui/qunit/sinon-qunit-bridge": {
+            deps: ["sap/ui/thirdparty/qunit", "sap/ui/thirdparty/sinon-4"]
+        }        
+    }
+});
+
+sap.ui.require([
+    "sap/ui/thirdparty/qunit-2",
+    "sap/ui/qunit/qunit-junit",
+    "sap/ui/qunit/qunit-coverage"
+], function (QUnit) {
+    'use strict';
+    QUnit.config.autostart = false;
```

Comment:
After understanding the timing issue with the async loader used for testing, this code here will not be safe that QUnit doesn't start automatically. When requiring QUnit with `sap.ui.require` or `sap.ui.define` it will always automatically start unless the `QUnit.config.autostart` is set to false before QUnit is loaded.

Two solution options:
1. Use the script tags again
2. Set autostart to false before loading QUnit

```js
sap.ui.loader.config({
    shim: {
        "sap/ui/qunit/qunit-junit": {
            deps: ["sap/ui/thirdparty/qunit"]
        },
        "sap/ui/qunit/qunit-coverage": {
            deps: ["sap/ui/thirdparty/qunit"]
        },
        "sap/ui/thirdparty/sinon-qunit": {
            deps: ["sap/ui/thirdparty/qunit", "sap/ui/thirdparty/sinon"]
        },
        "sap/ui/qunit/sinon-qunit-bridge": {
            deps: ["sap/ui/thirdparty/qunit", "sap/ui/thirdparty/sinon-4"]
        }        
    }
});

window.QUnit ??= {};
window.QUnit.config ??= {};
window.QUnit.config.autostart ??= false;

sap.ui.define([
    "sap/ui/thirdparty/qunit",
    "sap/ui/qunit/qunit-junit",
    "sap/ui/qunit/qunit-coverage"
], function () {
```

Reply:
Solution 2 (using QUnit preconfig) crashes ui5-test-runner. I'm in contact with @ArnaudBuchholz about that.

Reply:
Thanks!